gdx/src/com/badlogic/gdx/graphics/g2d/DistanceFieldBitmapFontCache.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright 2011 See AUTHORS file.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.graphics.g2d;
+
+import com.badlogic.gdx.graphics.glutils.ShaderProgram;
+
+
+/** 
+ * Caches and renders distance field font with custom shader. 
+ * Inspired by https://github.com/libgdx/libgdx/wiki/Distance-field-fonts
+ * @author Florian Falkner
+ */
+public class DistanceFieldBitmapFontCache extends BitmapFontCache {
+	private static ShaderProgram shader;
+	
+	/*
+	 * Needs to be done in a static block, otherwise doesn't work
+	 */
+	static {
+		shader = createDistanceFieldShader();
+	}
+	
+	public DistanceFieldBitmapFontCache (BitmapFont font, boolean integer) {
+		super(font, integer);
+	}
+
+	public DistanceFieldBitmapFontCache (BitmapFont font) {
+		super(font);
+	}
+	
+	@Override
+	public void draw (Batch spriteBatch) {
+		spriteBatch.setShader(shader);
+		shader.setUniformf("u_smoothing", getFont().getDistanceFieldSmoothing() * getFont().getScaleX());
+		
+		super.draw(spriteBatch);
+		spriteBatch.setShader(null);
```

Comment:
What if the user had already called setShader with something other than the default? This will blow that away. Should probably add a getShader method to Batch and then set it back to the previously set shader.